specs/waku/waku-rpc-api.md

```diff
@@ -0,0 +1,410 @@
+---
+
+## Table of Contents
+
+1. [Abstract](#abstract)
+2. [Wire Protocol](#wire-protocol)
+    1. [Transport](#transport)
+    1. [Objects](#objects)
+    1. [Methods](#methods)
+3. [Copyright](#copyright)
+
+## Abstract
+
+In this specification we describe the RPC API that Waku nodes SHOULD adhere to. The unified API allows clients to easily
+be able to connect to any node implementation.
+
+## Introduction
+
+This API is based off the [Whisper V6 RPC API](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API).
+
+## Wire Protocol
+
+### Transport
+
+Nodes SHOULD expose a [JSON RPC](https://www.jsonrpc.org/specification) API that can be publicly accessed. The JSON RPC version SHOULD be `2.0`. Below is an example request:
+
+```json
+{
+  "jsonrpc":"2.0",
+  "method":"shh_version",
+  "params":[],
+  "id":1
+}
+```
+
+#### Fields
+
+| Field     | Description                                         |
+| --------- | --------------------------------------------------- |
+| `jsonrpc` | Contains the used JSON RPC version (`Default: 2.0`) |
+| `method`  | Contains the JSON RPC method that is being called   |
+| `params`  | An array of parameters for the request              |
+| `id`      | The request ID                                      |
+
+### Objects
+
+In this section you will find objects used throughout the JSON RPC API.
+
+#### Message
+
+The message object represents a Waku message. Below you will find the description of the attributes contained in the message object.
+
+| Field | Type | Description |
+| ----: | :--: | ----------- |
+| `sig` | string | Public Key that signed this message |
+| `recipientPublicKey` | string | The recipients public key |
+| `ttl` | number | Time-to-live in seconds |
+| `timestamp` | number | Unix timestamp of the message generation |
+| `topic` | string | 4 bytes, the message topic |
+| `payload` | string | Decrypted payload |
+| `padding` | string | Optional padding, byte array of arbitrary length |
+| `pow` | number | The proof of work value |
+| `hash` | string | Hash of the enveloped message |
+
+#### Filter
+
+The filter object represents filters that can be applied to retrieve messages. Below you will find the description of the attributes contained in the filter object.
+
+| Field | Type | Description |
+| ----: | :--: | ----------- |
+| `symKeyID` | string | ID of the symmetric key for message decryption |
+| `privateKeyID` | string | ID of private (asymmetric) key for message decryption |
+| `sig` | string | Public key of the signature |
+| `minPow` | number | Minimal PoW requirement for incoming messages |
+| `topics` | array | Array of possible topics, this can also contain partial topics |
+| `allowP2P` | boolean | Indicates if this filter allows processing of direct peer-to-peer messages |
+
+All fields are optional, however `symKeyID` or `privateKeyID` must be present, it cannot be both. Additionally, the `topics` field is only optional when an asymmetric key is used.
+
+### Methods
```

Comment:
What accounts for the difference in Nim and Go impl? Is this the minimal subset? Are there yet undocumented ones that we want here?

Reply:
So this is based off of whisper v6 which matches what is in Waku Nim, I am not sure on the go implementation.

Reply:
FYI https://github.com/status-im/status-go/blob/develop/waku/api.go has all the RPC methods currently supported by `status-go`

I notice that `Version()` is missing